### PR TITLE
Use proptest instead of quickcheck for zcash_history tests.

### PR DESCRIPTION
**Pull request by nuttycom**
Part of #757; resolves duplicate `rand-core` dependencies.

Best viewed ignoring whitespace changes.

### CODE REVIEW

File: zcash_history/src/tree.rs [hunk 1]
```diff
@@ -640,100 +639,88 @@ mod tests {
...
         }
 
-        fn parity_with_truncate(add: u32, delete: u32) -> TestResult {
+        #[test]
+        fn prop_parity_with_truncate(
+            add_and_delete in (0u32..=2048).prop_flat_map(
+                |add| (Just(add), 0..=add)
+            )
+        ) {
+            let (add, delete) = add_and_delete;
```
Comment (str4d) [inline zcash_history/src/tree.rs:686]: I believe this works:
```suggestion
            (add, delete) in (0u32..=2048).prop_flat_map(
                |add| (Just(add), 0..=add)
            )
        ) {
```

File: zcash_history/src/tree.rs [hunk 2]
```diff
@@ -640,100 +639,88 @@ mod tests {
...
-                for _ in 0..delete {
-                    tree.truncate_leaf().expect("Failed to truncate");
-                }
+        #[test]
+        fn prop_stored_length(
+            add_and_delete in (0u32..=2048).prop_flat_map(
+                |add| (Just(add), 0..=add)
+            )
+        ) {
+            let (add, delete) = add_and_delete;
```
Comment (str4d) [inline zcash_history/src/tree.rs:712]:
```suggestion
            (add, delete) in (0u32..=2048).prop_flat_map(
                |add| (Just(add), 0..=add)
            )
        ) {
```

**Review by str4d — APPROVED** [review]
utACK

File: zcash_history/src/tree.rs (hunk 1 above)
Comment (nuttycom) [reply zcash_history/src/tree.rs:686]: I tried that first; the macro didn't like it.

Comment (str4d) [reply zcash_history/src/tree.rs:686]: Hmm, maybe because of the `prop_flat_map`.

**Pull request merged**